### ISSUE TITLE
Ant: WARNING: Unexpected error during parsing of the pom '...'

### ISSUE THREAD

**Issue by chilloutman** [dependency-check/DependencyCheck#141]
Running the ant task causes several warning to be printed:

```
WARNING: Unexpected error during parsing of the pom 'META-INF/maven/.../pom.xml'
```

That's not unexpected, since we don't use maven at all and some of the jars don't have a pom.xml. But my question is: Does the check still happen or is the pom.xml required?

**Comment by jeremylong** [comment]
Sorry for the delayed response. This is due to one or more JAR files
containing POM files that for whatever reason cannot be parsed. If you have
an example JAR file that contains one of the pom files I would love to see
it so that I can fix the parser.

Best Regards,

Jeremy

**Comment by chilloutman** [comment]
Several apache-commons-*.jar are causing the issue. Ive uploaded the contents of a pom.xml here:
https://gist.github.com/chilloutman/80b8181c63db345a1ebe

Also if you download a Tomcat server release and run the task on that it should find several jars with the same issue.

**Comment by chilloutman** [comment]
Actually, I just found something more concrete in the logs:

```
SEVERE: Unable to load parser. See the log for more details.
Jul 30, 2014 2:02:42 PM org.owasp.dependencycheck.analyzer.JarAnalyzer <init>
FINE: null
javax.xml.bind.JAXBException: "org.owasp.dependencycheck.jaxb.pom.generated" doesnt contain ObjectFactory.class or jaxb.index
        at com.sun.xml.internal.bind.v2.ContextFactory.createContext(ContextFactory.java:186)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:597)
        at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:128)
        at javax.xml.bind.ContextFinder.find(ContextFinder.java:290)
        at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:372)
        at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:337)
        at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:244)
        at org.owasp.dependencycheck.analyzer.JarAnalyzer.<init>(JarAnalyzer.java:172)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:39)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
        at java.lang.Class.newInstance0(Class.java:357)
        at java.lang.Class.newInstance(Class.java:310)
        at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:356)
        at java.util.ServiceLoader$1.next(ServiceLoader.java:428)
        at org.owasp.dependencycheck.Engine.loadAnalyzers(Engine.java:132)
        at org.owasp.dependencycheck.Engine.<init>(Engine.java:110)
        at org.owasp.dependencycheck.taskdefs.DependencyCheckTask.execute(DependencyCheckTask.java:893)
        at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
        at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:597)
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
        at org.apache.tools.ant.Task.perform(Task.java:348)
        at org.apache.tools.ant.Target.execute(Target.java:435)
        at org.apache.tools.ant.Target.performTasks(Target.java:456)
        at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
        at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
        at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
        at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
        at org.apache.tools.ant.Main.runBuild(Main.java:851)
        at org.apache.tools.ant.Main.startAnt(Main.java:235)
        at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
        at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
```

This appears quite early, before the actual analysis starts.
Am I missing a dependency?

**Comment by stevespringett** [comment]
Make sure you follow the instructions outlined here: http://jeremylong.github.io/DependencyCheck/dependency-check-ant/

Also, ensure that dependency-check-ant-x.x.x.jar is in the classpath BEFORE starting Ant.

**Comment by chilloutman** [comment]
Thank you @stevespringett. I've tried adding the jar externally using `ant -lib '...'` and the warnings went away.

Is there no way to have the ant build be self-contained? I'd rather not have to install the jar or inject it via `-lib` everywhere.

**Comment by chilloutman** [comment]
This is the best solution I've come up this so far:

``` xml
<!-- Run owasp-dependency-check-internal in a forked process with the jar already in the ant classpath. -->
<java
    classname="org.apache.tools.ant.launch.Launcher"
    fork="true"
    failonerror="true"
    dir="${basedir}"
    taskname="startAnt"
>
    <classpath>
        <pathelement location="${ant.home}/lib/ant-launcher.jar"/>
        <fileset dir="${lib.dir}/dependency-check-ant" includes="*.jar"/>
    </classpath>
    <arg value="-buildfile"/>
    <arg file="${ant.file}"/>
    <arg value="owasp-dependency-check-internal"/>
</java>
```

This calls the `owasp-dependency-check-internal` target with an updated classpath. This way the target can be called without providing the jar and it takes care of the classpath by forking.

**Comment by jeremylong** [comment]
This has been fixed in the current snapshot. The ant task should be completely stand alone now. If it is not installed in the ant/lib directory you will need to add the classpath node to the taskdef.

--Jeremy

**Comment by jeremylong** [comment]
Upon further investigation, this problem appears to have affected the Jenkins plugin also. Regardless, it has been fixed in 1.2.4-SNAPSHOT which will be released in the next few days.
